src/components/results/body/index.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { TREE_VIEW, JSON_VIEW } from '../results-view-selector';
+import JSONTree from 'react-json-tree';
+import { escapeLikeJSON, stringify } from '../utils';
+
+import './style.css';
+
+class RequestBody extends React.Component {
+
+	render() {
+		const { response, view } = this.props;
+		const expandFalse = () => false;
+		const getItemString = ( type, data, itemType, itemString ) => (
+			/* eslint-disable react/no-danger */
+			<span
+				className="collapsed-content"
+				dangerouslySetInnerHTML={ { __html: `${ itemString } <span class="content">${ stringify( data ) }</span>` } }
+			/>
+			/* eslint-enable react/no-danger */
+		);
+
+		const jsonTheme = {
+			extend: 'default',
+			tree: {
+				MozUserSelect: 'text',
+				WebkitUserSelect: 'text',
+			},
+			nestedNodeItemString: ( node, expandedNodes, type, expanded ) => {
+				return {
+					className: expanded ? 'expanded' : 'collapsed',
+				};
+			},
+		};
+
+		const valueRenderer = value => {
+			if ( typeof value === 'string' ) {
+				const valueWithoutQuotes = value.replace( /^"|"$/g, '' );
+				let valueEscaped;
+				if ( valueWithoutQuotes === value ) {
+					// Probably a boolean or number
+					valueEscaped = escapeLikeJSON( value );
+				} else {
+					// String
+					valueEscaped = '"' + escapeLikeJSON( valueWithoutQuotes ) + '"';
+				}
+				return (
+					<span className="expanded-value">{valueEscaped}</span>
+				);
+			}
+
+			return value;
+		};
+
+		return (
+			<div className="response">
+				{response && response.body && view === TREE_VIEW &&
+				<JSONTree
+					theme={ jsonTheme }
+					data={ response.body }
+					shouldExpandNode={ expandFalse }
```

Comment:
if we define `expandFalse` inside of `render` it will always be recreated on every render, triggering unessential re-renders. if we define it in the module scope or on the class during initialization, however, it will remain the same reference and avoid those failures to detect equality through the shallow-equality test.